azure-functions/acbs-function/helpers/activity.js

```diff
@@ -0,0 +1,108 @@
+const { getUnixTime } = require('date-fns');
+const CONSTANTS = require('../constants');
+const api = require('../api-dtfs-cental-api');
+
+let deal;
+let record;
+let acbsResponse;
+let dealId;
+let updateRecord;
+
+const labelCase = (label) => label.charAt(0).toUpperCase() + label.substring(1).toLowerCase();
+
+const getTimestamp = () => getUnixTime(acbsResponse.headers ? new Date(acbsResponse.headers.date) : new Date());
+
+const getAuthor = () => {
+  const author = {
+    firstName: '',
+    lastName: '',
+    _id: '',
+  };
+
+  if (record === 'deal' || record === 'facility') {
+    author.firstName = deal.dealSnapshot.bank.name;
+    author._id = deal.dealSnapshot.bank.id;
+  }
+
+  return author;
+};
+
+const getLabel = () => {
+  const dealType = deal.dealSnapshot.details
+    ? labelCase(deal.dealSnapshot.details.submissionType)
+    : labelCase(deal.dealSnapshot.submissionType);
+  const text = updateRecord ? 'updated' : 'submitted';
+
+  if (record === 'deal') {
+    return `${dealType} ${text}`;
+  }
+  if (record === 'facility') {
+    return `Facility ${text}`;
+  }
+  return '';
+};
+
+const getComments = (role) => {
+  if (deal.dealSnapshot.dealType === CONSTANTS.PRODUCT.TYPE.BSS) {
+    const comments = deal.dealSnapshot.comments.filter((comment) => comment.user.roles.includes(role));
+    if (comments.length > 0) {
+      return comments[0].text;
+    }
+  }
+
+  if (deal.dealSnapshot.dealType === CONSTANTS.PRODUCT.TYPE.GEF) {
+    const comments = deal.dealSnapshot.comments.filter((comment) => comment.role === role);
+    if (comments.length > 0) {
+      return comments[0].comment;
+    }
+  }
+  return '';
+};
+
+const getDescription = () => {
+  if (record === 'deal') {
+    return getComments(CONSTANTS.ACTIVITY.ROLE.CHECKER);
+  }
+  return '';
+};
+
+const getObject = async () => {
+  let object = {};
+  try {
+    object = await api.getTfmDeal(dealId).then((data) => {
+      if (data && data._id) {
+        deal = data;
+        return {
+          type: CONSTANTS.ACTIVITY.TYPE.ACTIVITY,
+          timestamp: getTimestamp(),
+          author: getAuthor(),
+          text: getDescription(),
+          label: getLabel(),
+        };
+      }
+      return {};
+    });
+  } catch (error) {
+    console.error('Error creating activity object.', { error });
+  }
+  return object;
+};
+
+const add = (id, response, type, update = false) => {
+  if (id && response && type) {
+    dealId = id;
+    acbsResponse = response;
+    record = type;
+    updateRecord = update;
+    getObject().then((result) => {
+      const tfm = {
+        activities: result,
+      };
+      api.updateTfmDeal(dealId, tfm);
+    });
+  }
+};
+
+module.exports = {
+  add,
+};
```

Comment:
Are we able to add tests for these - I think there were challenges around adding tests to ACBS but what about unit tests?

Reply:
Test in durable functions, won't work as its hard to trigger them unless sending a deal object.

Reply:
But aren't the helper files pure JS, I was thinking could use jest for these?

Reply:
Oh yeah, true we can use Jest sorry mate I thought you were talking about E2E, will do.

Reply:
No worries E2E we shouldn't do for ACBS I don't think, ACBS technically has nothing to do with the E2E flow. 

Just concerned that we have a lot of helper/mapping functions for ACBS with nothing tested, will be great to make a start on it 👍